orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -614,22 +614,43 @@ def test_input_requires_grad_backward_creates_input_grad_as_required0(device):
 
     _, y2 = model(x1,x2)
     s2 = y2.sum()
-    s2.backward()
+    s2.backward()   # y1's gradient will be materialized.
     named_params = list(model.named_parameters())
     assert named_params[0][1].grad is None or torch.count_nonzero(named_params[0][1].grad) == 0
     assert named_params[1][1].grad is None or torch.count_nonzero(named_params[1][1].grad) == 0
     assert torch.count_nonzero(named_params[2][1].grad) > 0
     assert torch.count_nonzero(named_params[3][1].grad) > 0
 
+@pytest.mark.parametrize("device", ['cuda'])
+def test_loss_combines_two_outputs_with_dependency(device):
+
+    def run_step(model, x1, x2):
+        y1, y2 = model(x1, x2)
+        loss = y1.sum() + y2.sum()
+        loss.backward()
+
+    N, D_in, H, D_out = 32, 784, 500, 10
+    pt_model = NeuralNetMultiplePositionalArgumentsMultipleOutputs1(D_in, H, D_out).to(device)
```

Comment:
nit. better name, ...MultiOutputsWithDependency

Reply:
same applies to NeuralNetMultiplePositionalArgumentsMultipleOutputs0 
...MultiOutputsWithoutDependency